### PR TITLE
Sort package arguments in Dockerfile

### PR DESCRIPTION
**Pull request by bebehei**
This applies the Dockerfile best practices from dockerhub.

Also it adds `--no-cache` to the `apk add` calls. This has got the same effect
like `apk update`, but doesn't store the package lists in the container.

### CODE REVIEW

**Comment by bebehei** [comment]
Guys, thanks for those reviews. But why isn't travis reporting anything?

**Comment by pitkley** [comment]
@bebehei it is probably related to the recent move to an organization. [I have informed Daniel about it.](https://github.com/the-paperless-project/paperless/issues/470#issuecomment-464129353)

**Comment by danielquinn** [comment]
Sorry about this guys.  I'm having trouble with Travis (*why can't we all just use GitLab?) and haven't been able to figure out what the problem is.  Details are in #512.  I'll get it up & running as soon as I can.

**Comment by bebehei** [comment]
So, I made a force push now. It's building fine now.

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,16 @@ matrix:
         - python: "3.5"
         - python: "3.6"
         - python: "3.7-dev"
+        - env: BUILD_DOCKER=1
+          servces:
```
Comment (MasterofJOKers) [inline .travis.yml]: Is that typo here important?

File: Dockerfile [hunk 2]
```diff
@@ -12,11 +12,30 @@ COPY scripts/docker-entrypoint.sh /sbin/docker-entrypoint.sh
 ENV PAPERLESS_EXPORT_DIR=/export \
     PAPERLESS_CONSUMPTION_DIR=/consume
 
-
-RUN apk update --no-cache && apk add python3 gnupg libmagic libpq bash shadow curl \
```
Comment (MasterofJOKers) [inline Dockerfile]: Why is there no `apk update` needed anymore?

Comment (MasterofJOKers) [reply Dockerfile]: My bad. It's in the commit message :)

File: .travis.yml (hunk 1 above)
Comment (bebehei) [reply .travis.yml]: That's weird. That typo should actually prevent everything intended in this commit. But it still works.

Thanks for mentioning.

**Comment by bebehei** [comment]
Well, Travis status reports are fucked up again. Interestingly Travis builds the PR's commits, but it doesn't report the commit's status. [Build for 557d730](https://travis-ci.org/the-paperless-project/paperless/builds/495155225).

File: .travis.yml [hunk 3]
```diff
@@ -12,6 +12,17 @@ matrix:
...
         - python: "3.7-dev"
+        - env: BUILD_DOCKER=1
+          services:
+            - docker
+          before_install:
+            - true
+          install:
+            - true
+          script:
+            - docker build -t paperless .
```
Comment (Strubbl) [inline .travis.yml]: should we include a --pull here?

`docker build --pull -t paperless .`

**Comment by bebehei** [comment]
Travis hasn't got any images preloaded. So it doesn't matter, the image will get pulled anyway

**Comment by pitkley** [comment]
I am going to close and reopen this PR, since that [seems to fix Travis](https://github.com/the-paperless-project/paperless/issues/512#issuecomment-469050805) (thanks for discovering this easy fix, @stgarf 👍).

**Comment by bebehei** [comment]
So guys, will you finally merge this now?

**Comment by stgarf** [comment]
@bebehei Looks like as soon as it has two "approved" reviews it will be merged. There was an issue with Travis not running CI on in-progress PRs during the "Great Travis Paperless Organization Migration of 2019&trade;" that's been mostly resolved as of today.

**Comment by jat255** [comment]
Thanks @stgarf for going through all these older and held-up issues. I don't have any experience running with Docker, so I'll have to leave this review to someone else.

**Pull request merged**